### DISCUSSION TITLE
How could I set the location of openblas/lapack when I compile mxnet from source?

### DISCUSSION THREAD

**Discussion by CoinCheung** [apache/incubator-mxnet#16587] [Q&A] [answered]
Hi, 

I installed openblas and lapack to /opt/openblas and /opt/lapack. When I compile mxnet from source I use this: 
```
mkdir -p build
cmake .. -DCMAKE_BUILD_TYPE=Release -GNinja -DCMAKE_MODULE_PATH=/opt/openblas/lib/cmake/openblas
```
Then I got the error: 
-- Intel(R) MKL: lib /root/build/incubator-mxnet/build/mklml/mklml_lnx_2019.0.5.20190502/lib/libmklml_intel.so
-- Found OpenMP_C: -fopenmp (found version "4.0") 
-- Found OpenMP_CXX: -fopenmp (found version "4.0") 
-- Found OpenMP: TRUE (found version "4.0")  
-- OpenMP lib: /root/build/incubator-mxnet/build/mklml/mklml_lnx_2019.0.5.20190502/lib/libiomp5.so
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- VTune profiling environment is unset
-- Found Git: /usr/bin/git (found version "2.7.4") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found CUDA: /usr/local/cuda (found version "10.0") 
-- Found NVTX: /usr/local/cuda/include  
-- Found NVTX (include: /usr/local/cuda/include, library: /usr/local/cuda/lib64/libnvToolsExt.so)
-- Could NOT find MKL (missing: MKL_INCLUDE_DIR MKL_RT_LIBRARY) 
-- Could not find OpenBLAS include. Turning OpenBLAS_FOUND off
-- Could not find OpenBLAS lib. Turning OpenBLAS_FOUND off
CMake Error at cmake/Modules/FindOpenBLAS.cmake:82 (MESSAGE):
  Could not find OpenBLAS
Call Stack (most recent call first):
  cmake/ChooseBlas.cmake:42 (find_package)
  CMakeLists.txt:333 (include)

-- Configuring incomplete, errors occurred!
See also "/root/build/incubator-mxnet/build/CMakeFiles/CMakeOutput.log".
See also "/root/build/incubator-mxnet/build/CMakeFiles/CMakeError.log".

How could I let cmake find my openblas ?

**Comment by haojin2** [comment]
Does the CMake run without explicitly specifying your openblas and lapack locations?

**Comment by CoinCheung** [comment]
For lapack, I modified the path at this line, which does not make sense
https://github.com/apache/incubator-mxnet/blob/c3395ca60b20f4388dd76746696497148b82fc80/make/config.mk#L134

For openblas, I have to change the folder name from /opt/openblas to /opt/OpenBLAS to make it work. 

Would you please show me how I could specify the locations explicitly ?

**Comment by szha** [comment, marked as answer]
Locating openblas should happen automatically with cmake. You can override with `OpenBLAS_INCLUDE_DIR ` variable https://github.com/apache/incubator-mxnet/blob/master/cmake/ChooseBlas.cmake#L43
